src/LuigisBoxBundle/ValueObject/SearchUrlBuilder.php

```diff
@@ -133,6 +137,47 @@ public function setFilters(array $filters): self
         return $this;
     }
 
+    public function addMustFilter(string $key, string $value): self
+    {
+        $this->mustFilters[$key] = $this->mustFilters[$key] ?? [];
+        if (\is_string($this->mustFilters[$key])) {
+            $this->mustFilters[$key] = [$this->mustFilters[$key]];
+        }
+        $this->mustFilters[$key][] = $value;
+
+        $this->mustFilters[$key] = array_unique($this->mustFilters[$key]);
+
+        return $this;
+    }
+
+    public function setMustFilters(array $filters): self
+    {
+        if (\count($this->mustFilters ?? []) > 0) {
+            throw new \LogicException('You already have the must filters set. Use resetMustFilters() method to clear them first.');
+        }
+
+        $keys = array_keys($filters);
+        Assert::allString($keys, 'All must filters keys must be string.');
```

Comment:
Perhaps filters could be objects, with 'must' flag and it could validate itself? I think this code could be a lot clearer this way.

Reply:
I am not sure about this. I need to set 'simply filter' and 'must filter' at the same urlBuilder. These filters are separate elements. I need to set for example 'f[]:size:40' and 'f_must[]:size:40'.